From 13457538e065e2e0f9990999efe96e6c4d85da86 Mon Sep 17 00:00:00 2001
From: Yuantian Tang <andy.tang@nxp.com>
Date: Tue, 28 Oct 2025 23:51:08 -0700
Subject: [PATCH 49/56] LF-15768: arm64: dts: imx91-11x11-frdm-imx91s: add DT
 to  support Tianma TM050RDH03 panel

Since FRDM-IMX91 and FRDM-IMX91S have similar design on parallel display.
Move the common code from imx91-11x11-frdm-tianma-wvga-panel.dtso to common dtsi.
Both boards use the common dtsi code to support the TM050RDH03 panel.

Signed-off-by: Joseph Guo <qijian.guo@nxp.com>
Acked-by: Liu Ying <victor.liu@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |   3 +
 ...1-11x11-frdm-imx91s-tianma-wvga-panel.dtso |   6 +
 ...1-11x11-frdm-tianma-wvga-panel-common.dtsi | 162 ++++++++++++++++++
 .../imx91-11x11-frdm-tianma-wvga-panel.dtso   | 158 +----------------
 4 files changed, 172 insertions(+), 157 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx91-11x11-frdm-imx91s-tianma-wvga-panel.dtso
 create mode 100644 arch/arm64/boot/dts/freescale/imx91-11x11-frdm-tianma-wvga-panel-common.dtsi

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 74d2e1c97ee8..1f7b33c6d456 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -525,6 +525,9 @@ dtb-$(CONFIG_ARCH_MXC) += imx91-11x11-frdm-imx91s.dtb \
 			  imx91-11x11-frdm-imx91s-ld.dtb \
 			  imx91-11x11-frdm-imx91s-mt9m114.dtb
 
+imx91-11x11-frdm-imx91s-tianma-wvga-panel-dtbs := imx91-11x11-frdm-imx91s.dtb imx91-11x11-frdm-imx91s-tianma-wvga-panel.dtbo
+dtb-$(CONFIG_ARCH_MXC) += imx91-11x11-frdm-imx91s-tianma-wvga-panel.dtb
+
 imx93-11x11-evk-pmic-pf0900-dtbs := imx93-11x11-evk.dtb imx93-11x11-evk-pmic-pf0900.dtbo
 imx93-11x11-evk-pmic-pf0900-root-dtbs := imx93-11x11-evk-root.dtb imx93-11x11-evk-pmic-pf0900.dtbo
 imx93-11x11-evk-pmic-pf0900-flexio-i2c-dtbs := imx93-11x11-evk-flexio-i2c.dtb imx93-11x11-evk-pmic-pf0900.dtbo
diff --git a/arch/arm64/boot/dts/freescale/imx91-11x11-frdm-imx91s-tianma-wvga-panel.dtso b/arch/arm64/boot/dts/freescale/imx91-11x11-frdm-imx91s-tianma-wvga-panel.dtso
new file mode 100644
index 000000000000..40ac7d9f41a1
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx91-11x11-frdm-imx91s-tianma-wvga-panel.dtso
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "imx91-11x11-frdm-tianma-wvga-panel-common.dtsi"
diff --git a/arch/arm64/boot/dts/freescale/imx91-11x11-frdm-tianma-wvga-panel-common.dtsi b/arch/arm64/boot/dts/freescale/imx91-11x11-frdm-tianma-wvga-panel-common.dtsi
new file mode 100644
index 000000000000..60f52df040c1
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx91-11x11-frdm-tianma-wvga-panel-common.dtsi
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2025 NXP
+ */
+
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/pwm/pwm.h>
+#include "imx91-pinfunc.h"
+
+&{/} {
+	backlight: backlight {
+		compatible = "pwm-backlight";
+		pwms = <&tpm5 3 40000 PWM_POLARITY_INVERTED>;	/* 25KHz PWM */
+		brightness-levels = <0 100>;
+		num-interpolated-steps = <100>;
+		default-brightness-level = <80>;
+		power-supply = <&reg_vexp_5v>;
+	};
+
+	panel {
+		compatible = "tianma,tm050rdh03";
+		standby-gpios = <&gpio2 27 GPIO_ACTIVE_LOW>;	/* STBYB */
+		backlight = <&backlight>;
+		power-supply = <&reg_vexp_3v3>;
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&display_out>;
+			};
+		};
+	};
+};
+
+&gpio2 {
+	/* avoid power leakage */
+	disp-clk-hog {
+		gpio-hog;
+		gpios = <0 GPIO_ACTIVE_HIGH>;
+		output-low;
+	};
+
+	disp-de-hog {
+		gpio-hog;
+		gpios = <1 GPIO_ACTIVE_HIGH>;
+		output-low;
+	};
+
+	disp-vsync-hog {
+		gpio-hog;
+		gpios = <2 GPIO_ACTIVE_HIGH>;
+		output-low;
+	};
+
+	disp-hsync-hog {
+		gpio-hog;
+		gpios = <3 GPIO_ACTIVE_HIGH>;
+		output-low;
+	};
+};
+
+&iomuxc {
+	pinctrl_lcdif: lcdifgrp {
+		fsl,pins = <
+			MX91_PAD_GPIO_IO00__MEDIAMIX_DISP_CLK		0x31e
+			MX91_PAD_GPIO_IO01__MEDIAMIX_DISP_DE		0x31e
+			MX91_PAD_GPIO_IO02__MEDIAMIX_DISP_VSYNC		0x31e
+			MX91_PAD_GPIO_IO03__MEDIAMIX_DISP_HSYNC		0x31e
+			MX91_PAD_GPIO_IO04__MEDIAMIX_DISP_DATA0		0x31e
+			MX91_PAD_GPIO_IO05__MEDIAMIX_DISP_DATA1		0x31e
+			MX91_PAD_GPIO_IO06__MEDIAMIX_DISP_DATA2		0x31e
+			MX91_PAD_GPIO_IO07__MEDIAMIX_DISP_DATA3		0x31e
+			MX91_PAD_GPIO_IO08__MEDIAMIX_DISP_DATA4		0x31e
+			MX91_PAD_GPIO_IO09__MEDIAMIX_DISP_DATA5		0x31e
+			MX91_PAD_GPIO_IO10__MEDIAMIX_DISP_DATA6		0x31e
+			MX91_PAD_GPIO_IO11__MEDIAMIX_DISP_DATA7		0x31e
+			MX91_PAD_GPIO_IO12__MEDIAMIX_DISP_DATA8		0x31e
+			MX91_PAD_GPIO_IO13__MEDIAMIX_DISP_DATA9		0x31e
+			MX91_PAD_GPIO_IO14__MEDIAMIX_DISP_DATA10	0x31e
+			MX91_PAD_GPIO_IO15__MEDIAMIX_DISP_DATA11	0x31e
+			MX91_PAD_GPIO_IO16__MEDIAMIX_DISP_DATA12	0x31e
+			MX91_PAD_GPIO_IO17__MEDIAMIX_DISP_DATA13	0x31e
+			MX91_PAD_GPIO_IO18__MEDIAMIX_DISP_DATA14	0x31e
+			MX91_PAD_GPIO_IO19__MEDIAMIX_DISP_DATA15	0x31e
+			MX91_PAD_GPIO_IO20__MEDIAMIX_DISP_DATA16	0x31e
+			MX91_PAD_GPIO_IO21__MEDIAMIX_DISP_DATA17	0x31e
+			MX91_PAD_GPIO_IO27__GPIO2_IO27			0x31e
+		>;
+	};
+
+	pinctrl_lcdif_gpio: lcdifgpiogrp {
+		fsl,pins = <
+			MX91_PAD_GPIO_IO00__GPIO2_IO0			0x51e
+			MX91_PAD_GPIO_IO01__GPIO2_IO1			0x51e
+			MX91_PAD_GPIO_IO02__GPIO2_IO2			0x51e
+			MX91_PAD_GPIO_IO03__GPIO2_IO3			0x51e
+			MX91_PAD_GPIO_IO04__GPIO2_IO4			0x51e
+			MX91_PAD_GPIO_IO05__GPIO2_IO5			0x51e
+			MX91_PAD_GPIO_IO06__GPIO2_IO6			0x51e
+			MX91_PAD_GPIO_IO07__GPIO2_IO7			0x51e
+			MX91_PAD_GPIO_IO08__GPIO2_IO8			0x51e
+			MX91_PAD_GPIO_IO09__GPIO2_IO9			0x51e
+			MX91_PAD_GPIO_IO10__GPIO2_IO10			0x51e
+			MX91_PAD_GPIO_IO11__GPIO2_IO11			0x51e
+			MX91_PAD_GPIO_IO12__GPIO2_IO12			0x51e
+			MX91_PAD_GPIO_IO13__GPIO2_IO13			0x51e
+			MX91_PAD_GPIO_IO14__GPIO2_IO14			0x51e
+			MX91_PAD_GPIO_IO15__GPIO2_IO15			0x51e
+			MX91_PAD_GPIO_IO16__GPIO2_IO16			0x51e
+			MX91_PAD_GPIO_IO17__GPIO2_IO17			0x51e
+			MX91_PAD_GPIO_IO18__GPIO2_IO18			0x51e
+			MX91_PAD_GPIO_IO19__GPIO2_IO19			0x51e
+			MX91_PAD_GPIO_IO20__GPIO2_IO20			0x51e
+			MX91_PAD_GPIO_IO21__GPIO2_IO21			0x51e
+			MX91_PAD_GPIO_IO27__GPIO2_IO27			0x51e
+		>;
+	};
+
+	pinctrl_tpm5_ch3: tpm5ch3grp {
+		fsl,pins = <
+			MX91_PAD_GPIO_IO26__TPM5_CH3			0x31e
+		>;
+	};
+};
+
+&lcdif {
+	assigned-clock-rates = <300000000>, <30000000>, <333333334>, <125000000>;
+	status = "okay";
+};
+
+&media_blk_ctrl {
+	status = "okay";
+};
+
+&parallel_disp_fmt {
+	pinctrl-names = "default", "gpio";
+	pinctrl-0 = <&pinctrl_lcdif>;
+	pinctrl-1 = <&pinctrl_lcdif_gpio>;
+	fsl,interface-pix-fmt = "rgb666";
+	status = "okay";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@1 {
+			reg = <1>;
+
+			display_out: endpoint {
+				remote-endpoint = <&panel_in>;
+			};
+		};
+	};
+};
+
+&tpm5 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_tpm5_ch3>;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/imx91-11x11-frdm-tianma-wvga-panel.dtso b/arch/arm64/boot/dts/freescale/imx91-11x11-frdm-tianma-wvga-panel.dtso
index 468df937d06c..b7317133ea13 100644
--- a/arch/arm64/boot/dts/freescale/imx91-11x11-frdm-tianma-wvga-panel.dtso
+++ b/arch/arm64/boot/dts/freescale/imx91-11x11-frdm-tianma-wvga-panel.dtso
@@ -3,170 +3,14 @@
  * Copyright 2025 NXP
  */
 
-#include <dt-bindings/gpio/gpio.h>
-#include <dt-bindings/pwm/pwm.h>
-#include "imx91-pinfunc.h"
-
-/dts-v1/;
-/plugin/;
-
-&{/} {
-	backlight: backlight {
-		compatible = "pwm-backlight";
-		pwms = <&tpm5 3 40000 PWM_POLARITY_INVERTED>;	/* 25KHz PWM */
-		brightness-levels = <0 100>;
-		num-interpolated-steps = <100>;
-		default-brightness-level = <80>;
-		power-supply = <&reg_vexp_5v>;
-	};
-
-	panel {
-		compatible = "tianma,tm050rdh03";
-		standby-gpios = <&gpio2 27 GPIO_ACTIVE_LOW>;	/* STBYB */
-		backlight = <&backlight>;
-		power-supply = <&reg_vexp_3v3>;
-
-		port {
-			panel_in: endpoint {
-				remote-endpoint = <&display_out>;
-			};
-		};
-	};
-};
+#include "imx91-11x11-frdm-tianma-wvga-panel-common.dtsi"
 
 /* pin conflicts */
 &flexcan2 {
 	status = "disabled";
 };
 
-&gpio2 {
-	/* avoid power leakage */
-	disp-clk-hog {
-		gpio-hog;
-		gpios = <0 GPIO_ACTIVE_HIGH>;
-		output-low;
-	};
-
-	disp-de-hog {
-		gpio-hog;
-		gpios = <1 GPIO_ACTIVE_HIGH>;
-		output-low;
-	};
-
-	disp-vsync-hog {
-		gpio-hog;
-		gpios = <2 GPIO_ACTIVE_HIGH>;
-		output-low;
-	};
-
-	disp-hsync-hog {
-		gpio-hog;
-		gpios = <3 GPIO_ACTIVE_HIGH>;
-		output-low;
-	};
-};
-
-&iomuxc {
-	pinctrl_lcdif: lcdifgrp {
-		fsl,pins = <
-			MX91_PAD_GPIO_IO00__MEDIAMIX_DISP_CLK		0x31e
-			MX91_PAD_GPIO_IO01__MEDIAMIX_DISP_DE		0x31e
-			MX91_PAD_GPIO_IO02__MEDIAMIX_DISP_VSYNC		0x31e
-			MX91_PAD_GPIO_IO03__MEDIAMIX_DISP_HSYNC		0x31e
-			MX91_PAD_GPIO_IO04__MEDIAMIX_DISP_DATA0		0x31e
-			MX91_PAD_GPIO_IO05__MEDIAMIX_DISP_DATA1		0x31e
-			MX91_PAD_GPIO_IO06__MEDIAMIX_DISP_DATA2		0x31e
-			MX91_PAD_GPIO_IO07__MEDIAMIX_DISP_DATA3		0x31e
-			MX91_PAD_GPIO_IO08__MEDIAMIX_DISP_DATA4		0x31e
-			MX91_PAD_GPIO_IO09__MEDIAMIX_DISP_DATA5		0x31e
-			MX91_PAD_GPIO_IO10__MEDIAMIX_DISP_DATA6		0x31e
-			MX91_PAD_GPIO_IO11__MEDIAMIX_DISP_DATA7		0x31e
-			MX91_PAD_GPIO_IO12__MEDIAMIX_DISP_DATA8		0x31e
-			MX91_PAD_GPIO_IO13__MEDIAMIX_DISP_DATA9		0x31e
-			MX91_PAD_GPIO_IO14__MEDIAMIX_DISP_DATA10	0x31e
-			MX91_PAD_GPIO_IO15__MEDIAMIX_DISP_DATA11	0x31e
-			MX91_PAD_GPIO_IO16__MEDIAMIX_DISP_DATA12	0x31e
-			MX91_PAD_GPIO_IO17__MEDIAMIX_DISP_DATA13	0x31e
-			MX91_PAD_GPIO_IO18__MEDIAMIX_DISP_DATA14	0x31e
-			MX91_PAD_GPIO_IO19__MEDIAMIX_DISP_DATA15	0x31e
-			MX91_PAD_GPIO_IO20__MEDIAMIX_DISP_DATA16	0x31e
-			MX91_PAD_GPIO_IO21__MEDIAMIX_DISP_DATA17	0x31e
-			MX91_PAD_GPIO_IO27__GPIO2_IO27			0x31e
-		>;
-	};
-
-	pinctrl_lcdif_gpio: lcdifgpiogrp {
-		fsl,pins = <
-			MX91_PAD_GPIO_IO00__GPIO2_IO0			0x51e
-			MX91_PAD_GPIO_IO01__GPIO2_IO1			0x51e
-			MX91_PAD_GPIO_IO02__GPIO2_IO2			0x51e
-			MX91_PAD_GPIO_IO03__GPIO2_IO3			0x51e
-			MX91_PAD_GPIO_IO04__GPIO2_IO4			0x51e
-			MX91_PAD_GPIO_IO05__GPIO2_IO5			0x51e
-			MX91_PAD_GPIO_IO06__GPIO2_IO6			0x51e
-			MX91_PAD_GPIO_IO07__GPIO2_IO7			0x51e
-			MX91_PAD_GPIO_IO08__GPIO2_IO8			0x51e
-			MX91_PAD_GPIO_IO09__GPIO2_IO9			0x51e
-			MX91_PAD_GPIO_IO10__GPIO2_IO10			0x51e
-			MX91_PAD_GPIO_IO11__GPIO2_IO11			0x51e
-			MX91_PAD_GPIO_IO12__GPIO2_IO12			0x51e
-			MX91_PAD_GPIO_IO13__GPIO2_IO13			0x51e
-			MX91_PAD_GPIO_IO14__GPIO2_IO14			0x51e
-			MX91_PAD_GPIO_IO15__GPIO2_IO15			0x51e
-			MX91_PAD_GPIO_IO16__GPIO2_IO16			0x51e
-			MX91_PAD_GPIO_IO17__GPIO2_IO17			0x51e
-			MX91_PAD_GPIO_IO18__GPIO2_IO18			0x51e
-			MX91_PAD_GPIO_IO19__GPIO2_IO19			0x51e
-			MX91_PAD_GPIO_IO20__GPIO2_IO20			0x51e
-			MX91_PAD_GPIO_IO21__GPIO2_IO21			0x51e
-			MX91_PAD_GPIO_IO27__GPIO2_IO27			0x51e
-		>;
-	};
-
-	pinctrl_tpm5_ch3: tpm5ch3grp {
-		fsl,pins = <
-			MX91_PAD_GPIO_IO26__TPM5_CH3			0x31e
-		>;
-	};
-};
-
-&lcdif {
-	assigned-clock-rates = <300000000>, <30000000>, <333333334>, <125000000>;
-	status = "okay";
-};
-
 /* pin conflicts */
 &lpspi3 {
 	status = "disabled";
 };
-
-&media_blk_ctrl {
-	status = "okay";
-};
-
-&parallel_disp_fmt {
-	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl_lcdif>;
-	pinctrl-1 = <&pinctrl_lcdif_gpio>;
-	fsl,interface-pix-fmt = "rgb666";
-	status = "okay";
-
-	ports {
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		port@1 {
-			reg = <1>;
-
-			display_out: endpoint {
-				remote-endpoint = <&panel_in>;
-			};
-		};
-	};
-};
-
-&tpm5 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_tpm5_ch3>;
-	status = "okay";
-};
-- 
2.34.1

